.github/workflows/reusable-api-e2e.yml

```diff
@@ -7,77 +7,40 @@ env:
 on:
   workflow_call:
     inputs:
-      test-e2e-affected:
-        description: 'detect if we should run e2e tests'
-        required: false
-        default: true
-        type: boolean
-      test-e2e-ee-affected:
-        description: 'detect if we should run e2e-ee tests'
-        required: false
-        default: true
-        type: boolean
-      test-unit:
-        description: 'detect if we should run unit tests'
-        required: false
-        default: true
-        type: boolean
       ee:
         description: 'use the ee version of api'
         required: false
         default: false
         type: boolean
-      job-name:
-        description: 'job name [options: novu/api-ee, novu/api]'
-        required: true
-        type: string
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  check_submodule_token:
-    name: Check if the secret exists or not.
-    runs-on: ubuntu-latest
-    outputs:
-      has_token: ${{ steps.secret-check.outputs.has_token }}
-    steps:
-      - name: Check if secret exists
-        id: secret-check
-        run: |
-          if [[ -n "${{ secrets.SUBMODULES_TOKEN }}" ]]; then
-            echo "::set-output name=has_token::true"
-          else
-            echo "::set-output name=has_token::false"
-          fi
-
-  # This workflow contains a single job called "build"
   e2e_api:
     name: Test E2E
     runs-on: ubuntu-latest
-    timeout-minutes: 80
+    timeout-minutes: 30
     permissions:
       contents: read
-      packages: write
       deployments: write
       id-token: write
-    needs: [check_submodule_token]
+      packages: write
     steps:
-      # checkout with submodules if token is provided
+      # Checkout with EE-submodule if token provided
       - uses: actions/checkout@v4
         name: Checkout with submodules
-        if: ${{ needs.check_submodule_token.outputs.has_token == 'true' && inputs.ee }}
```

Comment:
how do we know if this pr was created by a maintainer or a community contributor?  

**background:** we had an issue where, when a maintainer triggers the ci, the token is available. but when a community contributor triggers it, the token is missing, causing the ci to fail.

Reply:
Good callout. What about replacing this with checking for `secrets.SUBMODULES_TOKEN` existence?

Reply:
Isn't that the value of check_submodule_token.outputs.has_token?